draft-ietf-tls-tls13.md

```diff
@@ -2795,13 +2800,25 @@ specified for all records when processing early data records.
 Specifically, decryption failure of any 0-RTT record following
 an accepted "early_data" extension MUST produce a fatal
 "bad_record_mac" alert as per {{record-payload-protection}}.
+Implementations SHOULD determine the security parameters for the
+1-RTT phase of the connection entirely before processing the EncryptedExtensions
+and Finished, using those values solely to determine whether to
+accept or reject the "early_data" extension.
 
 [[TODO: How does the client behave if the indication is rejected.]]
 
+##### Processing Order
 
-##### Replay Properties
+Clients are permitted to "stream" 0-RTT data until they
+receive the ServerHello, only then sending the "end_of_early_data"
+alert. In order to avoid deadlock, when accepting "early_data",
+servers MUST process the client's Finished and then immediately
+send the ServerHello, rather than waiting for the client's
+"end_of_early_data" alert.
```

Comment:
Is the `ServerHello` the only message that a client needs in order to switch to application data keys?  Because otherwise we should allow clients to stream until they receive the server's `Finished`.